### DISCUSSION TITLE
Sort Order statuses in custom order

### DISCUSSION THREAD

**Discussion by elidrissidev** [OpenMage/magento-lts#2202] [Ideas]
### Description (*)
Order statuses are sorted alphabetically by status code by default. We have a store with around 30 Order statuses in total. Every time I wanna filter Orders with a status I have to scan the select box options looking for it (see screenshot below, text replaced with "Example"). The same problem occurs in our internal system where we list the statuses.

![Screen Shot 2021-09-21 at 9 49 26 AM (2)](https://user-images.githubusercontent.com/67818913/134142910-b4a177c1-16d5-4e56-b4a9-bc0473d95840.png)

I say we could allow specifying a custom order for Order statuses when you create them and order them based on that across the Admin panel. What do you guys think? I can work on this myself If some people will find it useful.

### Expected behavior (*)
Order statuses should be displayed across the Admin area with a custom order (a number) defined when you create them.

### Benefits
Easily finding the status you need by listing them in a custom order.

### Additional information
Optionally, we could also list them in the select box grouped by `state`?
![Screen Shot 2021-09-21 at 10 23 56 AM](https://user-images.githubusercontent.com/67818913/134146320-8327ccb0-19f2-4ec9-b89d-8a42f1df1d2a.png)

**Comment by kiatng** [comment]
One project that I handled has a few thousands complex statuses. The workflow, which can span up to a year, is managed by it. There is no issue on ordering because there are only a handful of possible next statuses for any given status; the next statuses are comma-delimited and are stored in a column and configurable in backend.

**Comment by elidrissidev** [comment]
@kiatng Are you talking about Magento or some other project?

**Comment by kiatng** [comment]
> @kiatng Are you talking about Magento or some other project?

Yes, it's M1/OM.

![image](https://user-images.githubusercontent.com/1106470/134267595-96a11bf9-c990-499a-a0a4-7a771aa0246f.png)

There are hundreds of admin users working on the workflow, available statuses are filtered based on roles, further reducing the number of statuses. Because the flow is linear, the user can revert to an earlier status:

![image](https://user-images.githubusercontent.com/1106470/134268736-21be5e34-3d9f-4658-a166-4dee2780fdef.png)

In this way, there is no need to list all statuses. Example above has more than 30 statuses in the dropdown, but the next statuses are only 4 in black text, the orange ones are the statuses that came before in chrono order.

That's just how I solved it and has been in production for more than 7 years. Hopefully some of the ideas are applicable in your case.

**Comment by elidrissidev** [comment]
@kiatng Hmm, it's an interesting solution if you have hundreds or thousands of statuses, but it's very unlikely that we'll ever have more than a 100 status tbh, so this will be an overkill for us. I see you also have a `position` column there (I'm guessing it's for the same purpose as what I'm suggesting) which will be enough for now, but then it depends on how common this problem is, If not many people are facing it, I don't want to force it.

**Comment by elidrissidev** [comment]
Closing as it doesn't seem interesting for many users.

**Comment by addison74** [comment]
I moved it to **Discussions > Ideas** because there are advantages. It is a major difference between closing a good idea because it was posted in Issues and has no feedback for a while and moving it to discussions in a separate section. In the Issue section it will remain closed forever and no one will be interested in it, but in the Discussions section it remains open forever and does not bother anyone. At any time an opinion is welcome.

**Reply by elidrissidev** [reply]
Thanks! I will probably end up just creating a small extension for it.
